Alerting docs: Meta monitoring for cloud (#74797)
grafana/grafana: docs/sources/alerting/monitor/_index.md

---
aliases:

>   - alerting/meta-monitoring/

Reviewer: I've taken to the convention of adding a comment a the end of each line that includes the absolute path that we are redirecting. It makes it a lot easier to test the redirects and make sure that they get updated when we move files around.

Author: not sure what you mean?

Reviewer: An example is in https://github.com/grafana/grafana/pull/74673/files#diff-3ed03901c3fa92bdbec5da7e20588f4dc3bd1e4e22ead03d84f0fe0a68189b82R3-R9

Author: ok cool done!

  - ./set-up/meta-monitoring/ # /docs/grafana/<GRAFANA_VERSION>/alerting/set-up/meta-monitoring/
canonical: https://grafana.com/docs/grafana/latest/alerting/monitor/
description: Meta monitoring
keywords:
  - grafana
  - alerting
  - meta-monitoring
labels:
  products:
    - enterprise
    - oss
menuTitle: Monitor
title: Meta monitoring
weight: 200
---

# Meta monitoring

Monitor your alerting metrics to ensure you identify potential issues before they become critical.

Meta monitoring is the process of monitoring your monitoring system and alerting when your monitoring is not working as it should.

In order to enable you to meta monitor, Grafana provides predefined metrics.

Identify which metrics are critical to your monitoring system (i.e. Grafana) and then set up how you want to monitor them.

You can use meta-monitoring metrics to understand the health of your alerting system in the following ways:

1. [Optional] Create a dashboard in Grafana that uses this metric in a panel (just like you would for any other kind of metric).
2. [Optional] Create an alert rule in Grafana that checks this metric regularly (just like you would do for any other kind of alert rule).

> 3. [Optional] Use the Explore module in Grafana.

Reviewer 2: To do what?

Author: You can use meta-monitoring metrics to understand the health of your alerting system in the following ways:

### Example

```yaml
- job_name: grafana
  honor_timestamps: true
  scrape_interval: 15s
  scrape_timeout: 10s
  metrics_path: /metrics
  scheme: http
  follow_redirects: true
  static_configs:
    - targets:
        - grafana:3000
```

### List of available metrics

The Grafana ruler, which is responsible for evaluating alert rules, and the Grafana Alertmanager, which is responsible for sending notifications of firing and resolved alerts, provide a number of metrics that let you observe them.

#### grafana_alerting_alerts

This metric is a counter that shows you the number of `normal`, `pending`, `alerting`, `nodata` and `error` alerts. For example, you might want to create an alert that fires when `grafana_alerting_alerts{state="error"}` is greater than 0.

#### grafana_alerting_schedule_alert_rules

This metric is a gauge that shows you the number of alert rules scheduled. An alert rule is scheduled unless it is paused, and the value of this metric should match the total number of non-paused alert rules in Grafana.

#### grafana_alerting_schedule_periodic_duration_seconds_bucket

> This metric is a histogram that shows you the time it takes to process an individual tick in the scheduler that evaluates alert rules. If the scheduler takes longer than 10 seconds to process a tick then pending evaluations will start to accumulate such that alert rules might later than expected.

Reviewer 2: what is a tick?

Author: oh this was existing text - can ignore:)

> This metric is a histogram that shows you the time it takes to process an individual tick in the scheduler that evaluates alert rules. If the scheduler takes longer than 10 seconds to process a tick then pending evaluations will start to accumulate such that alert rules might later than expected.

Reviewer 2: ```suggestion
This metric is a histogram that shows the time it takes to process an individual tick in the scheduler that evaluates alert rules. If the scheduler takes longer than 10 seconds to process a tick, then pending evaluations will accumulate such that alert rules might be later than expected.
```

> These metrics are not available at present in Grafana Cloud.

## Metrics for Mimir-managed alerts

To meta monitor Grafana Mimir-managed alerts, open source and on-premise users need a Prometheus/Mimir server, or another metrics database to collect and store metrics exported by the Mimir ruler.

#### rule_evaluation_failures_total

This metric is a counter that shows you the total number of rule evaluation failures.

## Metrics for Alertmanager

To meta monitor the Alertmanager, you need a Prometheus/Mimir server, or another metrics database to collect and store metrics exported by Alertmanager.

For example, if you are using Prometheus you should add a `scrape_config` to Prometheus to scrape metrics from your Alertmanager.

### Example

```yaml
- job_name: alertmanager
  honor_timestamps: true
  scrape_interval: 15s
  scrape_timeout: 10s
  metrics_path: /metrics
  scheme: http
  follow_redirects: true
  static_configs:
    - targets:
        - alertmanager:9093
```

### List of available metrics

> 

Reviewer 2: should have an intro here, like "The following sections..." otherwise you have stacked headings.

Author: thanks..updated
